metaflow/plugins/cards/card_datastore.py

```diff
@@ -0,0 +1,200 @@
+""" 
+
+"""
+
+from collections import namedtuple
+from hashlib import sha1
+from io import BytesIO
+import os
+import shutil
+
+from metaflow.datastore.local_storage import LocalStorage
+from metaflow.metaflow_config import (
+    DATASTORE_CARD_S3ROOT,
+    DATASTORE_CARD_LOCALROOT,
+    DATASTORE_LOCAL_DIR,
+    DATASTORE_CARD_SUFFIX,
+)
+
+from .exception import CardNotPresentException
+
+TEMP_DIR_NAME = "metaflow_card_cache"
+NUM_SHORT_HASH_CHARS = 5
+
+CardInfo = namedtuple("CardInfo", ["type", "hash"])
+
+
+def path_spec_resolver(pathspec):
+    run_id, step_name, task_id = None, None, None
```

Comment:
nit: consider something like `*splits, *[None]*(4-len(splits))` for this whole function.